var/spack/repos/builtin/packages/palace/package.py

```diff
@@ -44,84 +42,100 @@ class Palace(CMakePackage):
-    depends_on("petsc~int64", when="~int64")
-    depends_on("petsc+openmp", when="+openmp")
-    depends_on("petsc~openmp", when="~openmp")
-
-    depends_on("slepc", when="+slepc")
-    depends_on("arpack-ng+icb@develop", when="+arpack")
-    depends_on("gslib", when="+gslib")
-
     with when("+superlu-dist"):
+        depends_on("superlu-dist+shared", when="+shared")
+        depends_on("superlu-dist~shared", when="~shared")
         depends_on("superlu-dist+int64", when="+int64")
         depends_on("superlu-dist~int64", when="~int64")
         depends_on("superlu-dist+openmp", when="+openmp")
         depends_on("superlu-dist~openmp", when="~openmp")
 
     with when("+strumpack"):
         depends_on("strumpack+butterflypack+zfp+parmetis")
+        depends_on("strumpack+shared", when="+shared")
+        depends_on("strumpack~shared", when="~shared")
         depends_on("strumpack+openmp", when="+openmp")
         depends_on("strumpack~openmp", when="~openmp")
 
     with when("+mumps"):
         depends_on("mumps+metis+parmetis")
+        depends_on("mumps+shared", when="+shared")
+        depends_on("mumps~shared", when="~shared")
         depends_on("mumps+openmp", when="+openmp")
         depends_on("mumps~openmp", when="~openmp")
 
-    # Conflicts: Palace always builds its own internal MFEM
+    with when("+slepc"):
+        depends_on("slepc ^petsc+mpi+double+complex")
+        depends_on("petsc+shared", when="+shared")
+        depends_on("petsc~shared", when="~shared")
+        depends_on("petsc+int64", when="+int64")
+        depends_on("petsc~int64", when="~int64")
+        depends_on("petsc+openmp", when="+openmp")
+        depends_on("petsc~openmp", when="~openmp")
+
+    with when("+arpack"):
+        depends_on("arpack-ng+mpi+icb@develop")
+        depends_on("arpack-ng+shared", when="+shared")
+        depends_on("arpack-ng~shared", when="~shared")
+
+    # Conflicts: Palace always builds its own internal MFEM, GSLIB
     conflicts("^mfem", msg="Palace builds its own internal MFEM")
+    conflicts("^gslib", msg="Palace builds its own internal GSLIB")
 
     # More dependency variant conflicts
     conflicts("^hypre+int64", msg="Palace uses HYPRE's mixedint option for 64 bit integers")
-    conflicts("^petsc~double", msg="Palace requires PETSc with double precision")
-    conflicts("^petsc~complex", msg="Palace requires PETSc with complex numbers")
-    conflicts("^petsc+metis", msg="Palace requires PETSc without METIS")
-    conflicts("^petsc+hypre", msg="Palace requires PETSc without HYPRE")
-    conflicts("^petsc+superlu-dist", msg="Palace requires PETSc without SuperLU_DIST")
-    conflicts("^slepc+arpack", msg="Palace requires SLEPc without ARPACK")
     conflicts("^mumps+int64", msg="Palace requires MUMPS without 64 bit integers")
+    conflicts("^slepc+arpack", msg="Palace requires SLEPc without ARPACK")
+
+    # No install phase for Palace (always performed during build)
+    phases = ["cmake", "build"]
```

Comment:
Can we avoid overriding phases of `CMakePackages`
```suggestion
```
but override the install phase instead to be a no-op?

Reply:
Fixed in [1cb31ba](https://github.com/spack/spack/pull/38910/commits/1cb31baf0909a967639d4fd5ad2c510e1c506208).